datasets/bg/jud_declarations/crawler.py

```diff
@@ -0,0 +1,171 @@
+import re
+import pdfplumber
+from lxml.html import HtmlElement
+from rigour.mime.types import PDF
+from normality import slugify
+from typing import Dict, Generator, List, Optional
+
+from zavod import Context, helpers as h
+from zavod.logic.pep import categorise, OccupancyStatus
+from zavod.shed.trans import apply_translit_full_name, make_position_translation_prompt
+
+
+TRANSLIT_OUTPUT = {"eng": ("Latin", "English")}
+POSITION_PROMPT = prompt = make_position_translation_prompt("bul")
+
+
+def extract_judicial_declaration(context, url, doc_id_date) -> Dict[str, Optional[str]]:
+    """Extract name, role, and organization from the first page of a judicial declaration PDF."""
+
+    pdf_path = context.fetch_resource(f"{doc_id_date}.pdf", url)
+    context.export_resource(pdf_path, PDF)
+
+    extracted_data = {}
+    try:
+        with pdfplumber.open(pdf_path) as pdf:
+            first_page = pdf.pages[0]
+            full_text = first_page.extract_text() or ""
+
+            # Regex patterns for extracting relevant fields
+            patterns = {
+                # Capture up until a newline or end of text
+                "name": r"Име:\s*([А-Яа-яЁё\s-]+)(?=\n|$)",
+                "role": r"Длъжност:\s*([А-Яа-яЁё\s-]+)(?=\n|$)",
+                "organization": r"Орган на съдебната власт:\s*([А-Яа-яЁё\s,.-]+)(?=\n|$)",
+            }
+
+            for key, pattern in patterns.items():
+                match = re.search(pattern, full_text)
+                extracted_data[key] = match.group(1).strip() if match else None
+    except Exception as e:
+        context.log.warning(f"Skipping {pdf_path} due to error: {e}")
+        return {}
+
+    context.log.info(f"Extracted Data: {extracted_data}")
+    return extracted_data
+
+
+def parse_html_table(
+    table: HtmlElement,
+    skiprows: int = 0,
+    headers: List[str] = [],
+) -> Generator[Dict[str, HtmlElement], None, None]:
+
+    first_row = table.find(".//tr[1]")
+    actual_headers = [td.text_content().strip() for td in first_row.findall("./td")]
```

Comment:
(nit) do we need to do text processing here?

`len(headers) == len(first_row.findall("./td"))` seems like it should be enough and expresses what we're actually checking here.